server/src/main/java/org/opensearch/cluster/awarenesshealth/ClusterAwarenessHealth.java

```diff
@@ -0,0 +1,371 @@
+import org.opensearch.common.settings.ClusterSettings;
+import org.opensearch.common.settings.Settings;
+
+import java.io.IOException;
+import java.util.Collections;
+import java.util.HashMap;
+import java.util.Map;
+import java.util.Objects;
+
+/**
+ * Cluster state Awareness health information
+ *
+ */
+public class ClusterAwarenessHealth implements Writeable {
+
+    private int numberOfNodes;
+    private int numberOfDataNodes;
+    private int activeShards;
+    private int unassignedShards;
+    private int initializingShards;
+    private int relocatingShards;
+    private int activePrimaryShards;
+    private double activeShardsPercent;
+    private boolean hasDiscoveredClusterManager;
+    private ClusterHealthStatus status;
+    private final ClusterAwarenessAttributeHealth clusterAwarenessAttributeHealth;
+
+    private final String ACTIVE = "active";
+    private final String INITIALIZING = "initializing";
+    private final String RELOCATING = "relocating";
+
+    /**
+     * Creates cluster awareness health from cluster state.
+     *
+     * @param clusterState The current cluster state. Must not be null.
+     * @param clusterSettings the current cluster settings.
+     * @param awarenessAttributeName Name of awareness attribute for which we need to see the health
+     */
+    public ClusterAwarenessHealth(ClusterState clusterState, ClusterSettings clusterSettings, String awarenessAttributeName) {
+
+        setClusterHealth(clusterState);
+
+        clusterAwarenessAttributeHealth = generateAwarenessAttributeHealthMap(clusterState, clusterSettings, awarenessAttributeName);
+    }
+
+    private void setClusterHealth(ClusterState clusterState) {
+        ClusterStateHealth clusterStateHealth = new ClusterStateHealth(clusterState);
+        numberOfDataNodes = clusterStateHealth.getNumberOfDataNodes();
+        numberOfNodes = clusterStateHealth.getNumberOfNodes();
+        activeShards = clusterStateHealth.getActiveShards();
+        unassignedShards = clusterStateHealth.getUnassignedShards();
+        initializingShards = clusterStateHealth.getInitializingShards();
+        relocatingShards = clusterStateHealth.getRelocatingShards();
+        status = clusterStateHealth.getStatus();
+        activePrimaryShards = clusterStateHealth.getActivePrimaryShards();
+        activeShardsPercent = clusterStateHealth.getActiveShardsPercent();
+        hasDiscoveredClusterManager = clusterStateHealth.hasDiscoveredClusterManager();
+    }
+
+    private Map<String, Map<String, Integer>> generateShardAllocationPerNode(ClusterState clusterState) {
+
+        boolean activeShard;
+        boolean relocatingShard;
+        boolean initializingShard;
+        String nodeId = "";
+        String relocatingNodeId = "";
+        Map<String, Integer> nodeShardState;
+
+        Map<String, Map<String, Integer>> shardAllocationPerNode = new HashMap<>();
+
+        for (ShardRouting shard : clusterState.routingTable().allShards()) {
```

Comment:
As nodeMap is the main criteria to see which nodes are part of cluster and this will just have nodes having shards we will not miss those nodes in the full constructed output